btree/btree.c

```diff
@@ -230,10 +230,115 @@
  *
  * Deletion (DEL)
  * ..............
+ * There are basically 2 cases for deletion
+ * 1. No underflow after deletion
+ * 2. Underflow after deletion
+ *  a. Balance by borrowing key from sibling
+ *  b. Balance by merging with sibling
+ *    b.i. No underflow at internal node 
+ *    b.ii. Underflow at internal node
+ *      b.ii.A. Borrow key-pivot from sibling at internal node
+ *      b.ii.B. Merge with sibling at internal node
+ * @verbatim
+ *
+ *                             INIT
+ *                               |
+ *                               v
+ *                             SETUP<----------------+ 
+ *                               |                   |
+ *                               v                   |
+ *                             LOCKALL<------------+ |
+ *                               |                 | |
+ *                               v                 | |
+ *                             DOWN<-------------+ | |
+ *                               |               | | |
+ *                               v               | | |
+ *                        +--->NEXTDOWN-->LOCK-->CHECK
+ *                        |     |                 |       +->MOVEUP
+ *                        |     |                 v       |      |
+ *                        +LOAD-+               ACT---->RESOLVE<-+
+ *                                               |              |
+ *                                               v              |
+ *                                             CLEANUP<----------+
+ *                                                |
+ *                                                v
+ *                                              DONE
+ * @endverbatim
  *
+ * Phases Description: 
+ * step 1. NEXTDOWN: travese down by searching for given key till we reach leaf node 
+ * step 2. LOAD : load left , right sibling  only if there are chances of underflow at the node (i.e. number of keys == min)
```

Comment:
In case of variable length of key and/or variable length of value, the number of keys/values that an internal node or a left node can contain, are not constant. 
How do we define the minimal of key/val?

Reply:
In such cases, if we have node size defined , we can consider node size instead of number of keys to determine if underflow can occur